### ISSUE TITLE
Android proxy detector crashes if HTTPUpdateDelegator.Setup is not run

### ISSUE THREAD

**Issue by ruben-ivre** [Benedicht/BestHTTP-Issues#216]
The android proxy detector class will crash the app if `HTTPUpdateDelegator.Setup` is not run before using it. 
Since the `Setup` method is not public, I had to do a  workaround calling `SetThreadingMode` to thread and back to unity update to ensure it's run.

The problem is that the android proxy detector uses `IsMainThread`, but that won't work before `Setup`, causing a crash when trying to do `AndroidJNI.DetachCurrentThread` because is detecting incorrectly that we're not in the main thread and trying to detach from it.

Since the `Setup` method is pretty safe, you could ensure setup is done from inside `IsMainThread`, as is pretty critical if it fails.

**Comment by Benedicht** [comment]
The intended method for these cases is to call `HTTPManager.Setup();`. HTTPManager glues thogether the plugin and coordinates it.

`HTTPManager.Setup();` calls `HTTPUpdateDelegator.CheckInstance();`. `HTTPUpdateDelegator.CheckInstance();` creates and attaches the component, and finally in its `Update` method the `Setup` function will be called on Unity's main thread.

It's advised to call `HTTPManager.Setup();` in your startup code.

**Comment by ruben-ivre** [comment]
> The intended method for these cases is to call `HTTPManager.Setup();`. HTTPManager glues thogether the plugin and coordinates it.
> 
> `HTTPManager.Setup();` calls `HTTPUpdateDelegator.CheckInstance();`. `HTTPUpdateDelegator.CheckInstance();` creates and attaches the component, and finally in its `Update` method the `Setup` function will be called on Unity's main thread.
> 
> It's advised to call `HTTPManager.Setup();` in your startup code.

I'm sorry but that's not an acceptable solution in my case: even if I called `HTTPManager.Setup();` I would still need to wait one frames or more or ensure the `HTTPUpdateDelegator.Setup` is called in the Update or the app will do a hard crash, so I don't have any proper solution that allows me to do initialization and connect in a single frame.

**Comment by Benedicht** [comment]
Yeah, you're right. Thinking about making `HTTPUpdateDelegator.Setup()` public. Calling it on the Unity main thread sooner shouldn't cause any issue.

**Comment by ruben-ivre** [comment]
> Yeah, you're right. Thinking about making `HTTPUpdateDelegator.Setup()` public. Calling it on the Unity main thread sooner shouldn't cause any issue.

Sounds ok to me.
